### DISCUSSION TITLE
Remove Google dependencies

### DISCUSSION THREAD

**Discussion by gary-host-laptop** [monkeytypegame/monkeytype#2259] [Ideas]
Google is a threat to an open internet and it is the biggest company that profits from data collection and surveillance mechanisms, and a lot of the products/services they offer are proprietary spyware. It would be incredible and a huge step in favour of a more private, open and decentralized network if projects would stop relying on such a harmful entity.

Either hosting the dependencies locally, or using a more private and open source alternative would be fine, I know there are a lot of good alternatives out there, you can probably find the one that suits you best over AlternativeTo.

**Comment by CameronCT** [comment]
Alright, let's be a little realistic here. 

If we were to remove "Google Dependencies" we would have to essentially remove Node, NPM and a lot of the most crucial things that make websites like Monkeytype, Keyma.sh and pretty much any other site exist.

A lot of libraries may have been created by others, but Google is pretty much one of the big hitters powering the Javascript Ecosystem including their Angular framework.

**Reply by gary-host-laptop** [reply]
I am honestly not to keen on using JavaScript but I understand the fact that is being used here, I'm not saying they should rebuild the whole of the project JSless because I'm not even sure that's possible (I'm not a developer). What I meant with Google dependencies in this case is stuff that points to Google servers like Google Fonts, which is pretty easy to find alternatives for (at least I've created similar issues in a lot of projects and most of the maintainers agreed to remove them and replace them with libre software alternatives) and if they use Google Analytics for example which is the most intrusive of all, but I can't see it using uBlock Origin so I assume it's not present. 

IMO the topic was pretty clear, if I were to ask something like "rebuild the project using Rust" because I don't like JS, the answer would have pretty much been "build it yourself" because at that point is not asking for an improvement, but for the creation of something completely different.

As a good example, Lichess.org uses JS among other languages and yet it does not contain Google dependencies.
